src/app-layout/__tests__/runtime-drawers.test.tsx

```diff
@@ -632,4 +639,361 @@ describeEachAppLayout(({ theme, size }) => {
+        type: 'global',
+        mountContent: container => (container.textContent = 'global drawer content 2'),
+      });
+
+      const { globalDrawersWrapper } = await renderComponent(<AppLayout drawers={[testDrawer]} />);
+
+      expect(globalDrawersWrapper.findActiveDrawers()).toHaveLength(0);
+
+      awsuiPlugins.appLayout.openDrawer('local-drawer');
+
+      await delay();
+
+      expect(globalDrawersWrapper.findActiveDrawers()).toHaveLength(1);
+
+      awsuiPlugins.appLayout.openDrawer('global-drawer-1');
+
+      await delay();
+
+      expect(globalDrawersWrapper.findActiveDrawers()).toHaveLength(2);
+    });
+
+    test('does not do anything when openDrawer is called with active drawer id', async () => {
+      awsuiPlugins.appLayout.registerDrawer({
+        ...drawerDefaults,
+        id: 'local-drawer',
+        mountContent: container => (container.textContent = 'local-drawer content'),
+      });
+
+      const { globalDrawersWrapper } = await renderComponent(<AppLayout drawers={[testDrawer]} />);
+
+      expect(globalDrawersWrapper.findActiveDrawers()).toHaveLength(0);
+
+      awsuiPlugins.appLayout.openDrawer('local-drawer');
+
+      await delay();
+
+      expect(globalDrawersWrapper.findActiveDrawers()).toHaveLength(1);
+
+      awsuiPlugins.appLayout.openDrawer('local-drawer');
+
+      await delay();
+
+      expect(globalDrawersWrapper.findActiveDrawers()).toHaveLength(1);
+    });
+
+    test('should restore focus when a global drawer is closed', async () => {
+      awsuiPlugins.appLayout.registerDrawer({
+        ...drawerDefaults,
+        id: 'global-drawer-1',
+        type: 'global',
+        mountContent: container => (container.textContent = 'global drawer content 1'),
+      });
+
+      const { wrapper, globalDrawersWrapper } = await renderComponent(<AppLayout drawers={[testDrawer]} />);
+
+      wrapper.findDrawerTriggerById('global-drawer-1')!.focus();
+      wrapper.findDrawerTriggerById('global-drawer-1')!.click();
+      expect(globalDrawersWrapper.findDrawerById('global-drawer-1')!.getElement()).toBeInTheDocument();
+      globalDrawersWrapper.findCloseButtonByActiveDrawerId('global-drawer-1')!.click();
+      expect(globalDrawersWrapper.findDrawerById('global-drawer-1')).toBeNull();
+      expect(wrapper.findDrawerTriggerById('global-drawer-1')!.getElement()).toHaveFocus();
+    });
+
+    test('should keep global drawer in DOM tree if preserveInactiveContent is set to true', async () => {
+      awsuiPlugins.appLayout.registerDrawer({
+        ...drawerDefaults,
+        id: 'global-drawer-1',
+        type: 'global',
+        mountContent: container => (container.textContent = 'global drawer content 1'),
+        preserveInactiveContent: true,
+      });
+
+      const { wrapper, globalDrawersWrapper } = await renderComponent(<AppLayout drawers={[testDrawer]} />);
+
+      expect(globalDrawersWrapper.findDrawerById('global-drawer-1')!.getElement()).toBeInTheDocument();
```

Comment:
How hard would it be to change the implementation so `preserveInactiveContent` will actually to the "preservation"? E.g. on the initial render the content is not mounted yet, but it is _preserved_ after closing the drawer?

I think this is what our consumers want and this is why it is called "preserve" instead of "renderWhenHidden" or something similar.

Reply:
That's not really difficult to implement. However, what do our consumers want is an open question. From my perspective, rendering such tabs initially is a way to go, because they would have the full control over the content of the tab via the visibility callback. Let's discuss it further anyway.